### ISSUE TITLE
Argument spreading should work in math function calls

### ISSUE THREAD

**Issue by SUPERCILEX** [typst/typst#938]
```
$mat(range(2).map(_ => range(2)))$
```
This just reprs the list which is expected, though a little confusing.

```
$mat(..#range(2).map(_ => range(2)))$
```
This does the same thing, but puts `..` in the matrix.

```
$mat(#..range(2).map(_ => range(2)))$
```

**This doesn't compile, but should IMO.**

```
$mat(#(..range(2).map(_ => range(2))))$
```

**This compiles, but doesn't work even though it should.**

---

```
$#math.mat(..range(2).map(_ => range(2)))$ \
$mat(#(..range(2).map(_ => range(2))))$ \
$mat(..#range(2).map(_ => range(2)))$
```

![image](https://user-images.githubusercontent.com/9490724/233813748-8d86a47f-c52e-484c-9c29-7b7403ee29ff.png)

**Comment by laurmaedje** [comment]
I'd say that the second example is the one that should work as it would be consistent with how named arguments work in math.

**Comment by SUPERCILEX** [comment]
> mat(..#range

This one? If it's possible to tell the difference between that and a banal use of `..` since you're not in code mode, then sure.

**Comment by laurmaedje** [comment]
if you want literal dots, you could disambiguate with `.. #range` or `\..#range`

**Comment by SUPERCILEX** [comment]
Good point, then I agree with your proposal. It's what I tried first IIRC so should be the most natural.
